Mathlib/Probability/Martingale/OptionalSampling.lean

```diff
@@ -149,6 +149,8 @@ theorem condexp_stoppedValue_stopping_time_ae_eq_restrict_le (h : Martingale f 
     (hτ : IsStoppingTime ℱ τ) (hσ : IsStoppingTime ℱ σ) [SigmaFinite (μ.trim hσ.measurableSpace_le)]
     (hτ_le : ∀ x, τ x ≤ i) :
     μ[stoppedValue f τ|hσ.measurableSpace] =ᵐ[μ.restrict {x : Ω | τ x ≤ σ x}] stoppedValue f τ := by
+  have := LinearLocallyFiniteOrder.succOrder ι
```

Comment:
It seems a pity to introduce code like this in files like `OptionalSampling` which are definitely far outside the API boundary of order theory. 

If I undestand correctly, these lines are needed in order to synthesize `OrderTopology ι`. But `OrderTopology` is Prop-valued, isn't it? It contains no data. So maybe we should have an instance somewhere much higher up (maybe `Mathlib.Topology.Order.Basic` along the lines of the following:
```
instance {ι : Type*} 
    [LinearOrder ι] [LocallyFiniteOrder ι] [TopologicalSpace ι] [DiscreteTopology ι] : 
    OrderTopology ι := 
  letI := LinearLocallyFiniteOrder.succOrder ι
  letI := LinearLocallyFiniteOrder.predOrder ι
  inferInstance
```
Then the existing code in `OptionalSampling` will "just work".

Reply:
I think we should add this instance to `Topology/Instances/Discrete`. We would need to add `Order/SuccPred/LinearLocallyFinite` to the imports of that file, but it seems that would only mean a 2% import increase in that file.

Reply:
(See #20261, which is a PR which I made solely in order to trigger the PR import-testing script, since it was easier than working out how to run the script locally.)

Reply:
Looks good to me. @urkud Do you agree? If so, we can merge David's branch into this one.

Reply:
I have no preferences here.

Reply:
Merged and shaken.